src/Compiler/pars.fsy

```diff
@@ -5136,8 +5136,11 @@ tupleOrQuotTypeElements:
     { [ SynTupleTypeSegment.Type $1 ] }
 
 appTypeCon:
+  | moduleKeyword path %prec prec_atomtyp_path 
+    { SynType.LongIdent($2, true) }
+
```

Comment:
This doesn't work. FsYacc prints some related shift/reduce errors, so precedence would appear to be the problem. I'd appreciate some help here as I am out of my depth.

Reply:
It's likely much more changes are needed in LexFilter, as it considers `module` and `type` special keywords that act as error recovery synchronization tokens.